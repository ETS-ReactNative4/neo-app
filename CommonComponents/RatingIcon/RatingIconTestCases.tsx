import React from "react";
import RatingIcon from "./RatingIcon";
import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import { CONSTANT_thirdColor } from "../../constants/colorPallete";

const RatingIconTestCases: ITestCase[] = [
  {
    title: "Rating Icon",
    Component: <RatingIcon numOfStars={1} rating={0} />
  },
  {
    title: "Rating Icon Active State",
    Component: <RatingIcon numOfStars={1} rating={1} />
  },
  {
    title: "Rating Icon Custom Color",
    Component: (
      <RatingIcon numOfStars={1} rating={1} activeColor={CONSTANT_thirdColor} />
    )
  },
  {
    title: "Multiple Rating Icon",
    Component: <RatingIcon rating={4} numOfStars={5} />
  },
  {
    title: "Rating Icon Custom Color With Active State",
    Component: (
      <RatingIcon rating={3} numOfStars={5} activeColor={CONSTANT_thirdColor} />
    )
  }
];

export default RatingIconTestCases;
