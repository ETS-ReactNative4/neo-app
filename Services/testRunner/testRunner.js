import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
