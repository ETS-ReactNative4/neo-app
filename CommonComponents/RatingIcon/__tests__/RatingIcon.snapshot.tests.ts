import RatingIconTestCases from "../RatingIconTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

RatingIconTestCases.map(testRunner);
