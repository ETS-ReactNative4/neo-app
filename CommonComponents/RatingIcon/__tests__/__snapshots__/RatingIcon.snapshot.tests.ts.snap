// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Rating Icon 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255, 201, 51, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255, 201, 51, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255, 201, 51, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255, 201, 51, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(221, 221, 221, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`Rating Icon 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(221, 221, 221, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`Rating Icon Active State 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255, 201, 51, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`Rating Icon Custom Color 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255,40,114,1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`Rating Icon Custom Color With Active State 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255,40,114,1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255,40,114,1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(255,40,114,1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(221, 221, 221, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(221, 221, 221, 1)",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
